python-frontend/src/main/java/org/sonar/python/types/v2/ModuleType.java

```diff
@@ -60,7 +79,11 @@ public ModuleType(@Nullable String name, @Nullable ModuleType parent) {
 
   @Override
   public Optional<PythonType> resolveMember(String memberName) {
-    return Optional.ofNullable(members.get(memberName));
+    return Optional.ofNullable(members.get(memberName)).map(TypeWrapper::type);
+  }
+
+  public Optional<PythonType> resolveSubmodule(String submoduleName) {
+    return Optional.ofNullable(subModules.get(submoduleName)).map(TypeWrapper::type);
```

Comment:
Is it expected that `subModules` is only used in test code? Not counting adding and retrieving from the parents, it never leaves the module types.

Reply:
This is a good point. I've added 13d2386c59f360c387ac80f302f0c6ad1a723995 to instead consistently register submodules in the `subModule` field.
I've updated the strategy of `resolveMember` to first resolve actual members, then sub modules if no member is present. The behavior is not entirely accurate depending on the context, but I believe it is a decent enough approximation before we tackle SONARPY-2176.